lib/gcloud/vision/project.rb

```diff
@@ -76,11 +83,146 @@ def self.default_project
           Gcloud::GCE.project_id
       end
 
+      ##
+      # Returns a new image from the given source.
+      #
+      # Image files sent to the Google Cloud Vision API should not exceed 4 MB.
+      # For multiple images in a request, the Vision API currently imposes an 8
+      # MB per request limit.
+      #
+      # Note that an object in Google Cloud Storage is a single entity;
+      # permissions affect only that object. "Directory permissions" do not
+      # exist (though default bucket permissions do exist). Make sure the code
+      # which performs your request has access to that image.
+      #
+      # @see https://cloud.google.com/vision/docs/image-best-practices Best
+      #   Practices
+      #
+      # @param [Gcloud::Storage::File, IO, StringIO, String] source a Google
+      #   Cloud Storage URI of the form `gs://bucketname/path/to/image_filename`
+      #   or an instance of Gcloud::Storage::File; or, a string path (file
+      #   descriptor); or a File, IO, or StringIO instance
+      #
+      # @return [Image] an image for the Vision service
+      #
+      # @example With a Google Cloud Storage URI:
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   vision = gcloud.vision
+      #
+      #   image = vision.image "gs://bucket-name/path_to_image_object"
+      #
+      # @example With a local file path:
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   vision = gcloud.vision
+      #
+      #   image = vision.image "./acceptance/data/landmark.jpg"
+      #
       def image source
         return source if source.is_a? Image
         Image.from_source source, self
       end
 
+      ##
+      # Performs detection of Cloud Vision [features](https://cloud.google.com/vision/reference/rest/v1/images/annotate#Feature)
+      # on the given image(s). If no options for features are provided, **all**
+      # image detection features will be performed, with a default of `10`
+      # results for faces, landmarks, logos, and labels. If any feature option
+      # is provided, only the specified feature detections will be performed.
+      # Please review [Pricing](https://cloud.google.com/vision/docs/pricing)
+      # before use, as a separate charge is incurred for each feature performed
+      # on an image.
+      #
+      # Image files sent to the Google Cloud Vision API should not exceed 4 MB.
+      # For multiple images in a request, the Vision API currently imposes an 8
+      # MB per request limit.
+      #
+      # @see https://cloud.google.com/vision/docs/requests-and-responses Cloud
+      #   Vision API Requests and Responses
+      # @see https://cloud.google.com/vision/reference/rest/v1/images/annotate#AnnotateImageRequest
+      #   AnnotateImageRequest
+      # @see https://cloud.google.com/vision/docs/pricing Cloud Vision Pricing
+      #
+      # @param [Image, Object] images The image or images to annotate. This can
```

Comment:
The casing of the arguments are inconsistent. Some start with an upper case letter, some with a lower case letter.